src/searching/searching.py

```diff
@@ -1,14 +1,35 @@
+
+# ? O(n)
+
 def linear_search(arr, target):
     # Your code here
+    for i in range(len(arr)):
+        if arr[i] == target:
+            return i
+        
 
 
     return -1   # not found
```

Comment:
Again, a completely valid and perfect implementation of linear search. However, I just want to keep poking you on the use of the `range` function when `enumerate` can clean up the readability of your code:

```python
def linear_search(arr, target ):
    for index, item in enumerate(arr):
        if item == target:
            return index

    return -1   # not found
```

As you can see, this is almost identical to your code, but provides that wonderful "near english" python syntax.